### DISCUSSION TITLE
esp32-wrover-b from makerfabs.com with any spi based tft_espi supported display.

### DISCUSSION THREAD

**Discussion by memeticae** [Bodmer/TFT_eSPI#1743] [General]
Hello,

If you have an "ESP32-WROVER-B" development board (e.g. from makerfabs.com) , then let me help you not to become as frustrated as I was.

("Invalid header" errors, will be the first hurdle whilst trying to upload code when anything is connected to the "wrong"pins.)

SPI on this board is rather tricky, since psram and spiflash both use the available spi channels OUTSIDE of the "IO-matrix".
(In fact: gpio 6 through 11, 16 and 17 are better not used at all, according to the fine documentation @ https://www.espressif.com/sites/default/files/documentation/esp32-wrover-b_datasheet_en.pdf)

Therefore you should use the assigned ports, which you can see when you flip the board over:

#define TFT_MISO 19   (VMISO)  // mandatory
#define TFT_MOSI 23   (VMOSI)  // mandatory
#define TFT_SCLK 18    (VCLK)    // mandatory
#define TFT_CS   5       (VCS)       // mandatory

The: TFT_DC , TFT_RST and TOUCH_CS pins can be chosen arbitrarily. (Yet steer away from the aformentioned ones.)

Only thus could I get the 4" MHS ST7796S TFT display to come alive.
( ... although only @40Mhz, most likely because of jumper cables about 20cm long and "breadboarding".

I hope this information can help to save someone from the frustrations I had to endure...

Best regards &c.
Chris.

PS. 
Regarding the "HSPI and VSPI" discussion: If you look at the pinout of the chip itself, you will notice that the HSPI interface is on the horizontal edge, whilst the VSPI one occupies the vertical edge.)
;)

**Comment by jobitjoseph** [comment]
Your information is really helpful. Avoid using GPIO16 and GPIO17 with the wrover module

**Reply by Bodmer** [reply]
Thanks for this information.
